src/core/ddsi/src/q_entity.c

```diff
@@ -2793,7 +2811,18 @@ static void reader_qos_mismatch (struct reader * rd, dds_qos_policy_id_t reason)
   }
 }
 
-static bool topickind_qos_match_p_lock (struct ddsi_domaingv *gv, struct entity_common *rd, const dds_qos_t *rdqos, const type_identifier_t *rd_typeid, struct entity_common *wr, const dds_qos_t *wrqos, const type_identifier_t *wr_typeid, dds_qos_policy_id_t *reason)
+static bool topickind_qos_match_p_lock (
```

Comment:
I have been staring at this (as well as `qos_match_p`) and I don't quite like the set of parameters, but I'm not quite sure much can be done to improve the situation: passing a struct in would require the struct to be initialized, &c. So I'm just mentioning it in case you have a good suggestion 🙂.

Reply:
I didn't really find a better solution for this, maybe the planned refactoring of fields in (proxy) entities will help in reducing the number of parameters for these function calls.